Sprint 15 (Add bookings)

VitalyVL1 opened the pull request:
Добавлен функционал:
- Работа с БД
- Бронирование вещи
- Отзывы после окончания аренды.

VitalyVL1 committed:
Добавить подключение к БД

VitalyVL1 committed:
Перенести функционал предыдущего (14 спринта) на работу с БД

VitalyVL1 committed:
Добавить функционал бронирование вещей

VitalyVL1 committed:
Добавить функционал отзывов

VitalyVL1 committed:
code style

VitalyVL1 committed:
Добавлена проверка на доступность времени бронирования. Немного оптимизирован метод save в BookingServiceImpl

VitalyVL1 committed:
Изменить структуру по пакетам, кроме моделей, убрать пакеты с одним классом, для лучшей читаемости

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/booking/model/State.java:
```diff
@@ -0,0 +1,5 @@
+
+public enum State {
+    ALL, WAITING, APPROVED, REJECTED, CANCELED
+}
```
ALL, CURRENT, PAST, FUTURE, WAITING, REJECTED

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -1,12 +1,77 @@
+            @PathVariable @Valid @NonNull @Positive Long bookingId
+    ) {
+        log.info("Getting a booking for {}", bookingId);
+        return bookingService.findById(bookingId);
```
"Может быть выполнено либо автором бронирования, либо владельцем вещи…" нужно в сервисе сделать проверку

VitalyVL1 committed:
Перенести все настройки в application.yaml
Исправить в BookingServiceImpl:
метод findById (выводит информацию только для владельца вещи или для того кто забронировал)
методы findByBookerIdAndState, findByOwnerIdAndState (информация выводится согласно следующим состояниям - ALL, CURRENT, PAST, FUTURE, WAITING, REJECTED)
